server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/node/StarTreeNode.java

```diff
@@ -109,26 +110,30 @@ public interface StarTreeNode {
      * @throws IOException if an I/O error occurs while retrieving the child node
      */
     default StarTreeNode getChildForDimensionValue(Long dimensionValue) throws IOException {
-        return getChildForDimensionValue(dimensionValue, null);
+        return getChildForDimensionValue(dimensionValue, null, null);
```

Comment:
Lets pass `DimensionDataType.LONG::compare` here for the time being. 

Once, we remove this method it would all go away together.